Ical.Net/Evaluation/RecurrencePatternEvaluator.cs

```diff
@@ -988,6 +990,6 @@ private static IDateTime MatchTimeZone(IDateTime dt1, IDateTime dt2)
 
         return dt1.IsUtc
             ? new CalDateTime(copy.AsUtc)
-            : new CalDateTime(copy.AsSystemLocal);
```

Comment:
Isn't the code in `MatchTimeZone(IDateTime dt1, IDateTime dt2)` completely useless?

Reply:
Its not the most elegant code, but it aims to adjust UNTIL, which usually is specified in UTC, to DTSTART's time zone, which usually isn't in UTC. This allows checking comparing the adjusted UNTIL with the generated recurrences without considering the time zone again. So the functionality is needed one way or the other.

Reply:
That's what I thought, but the way it is implemented looks like a noop. Then I removed the method invocation temporarily with no effect on unit tests. So this is just a hint re recurrences.

Reply:
I could easily be wrong, or relevant tests are missing. We'd need a test with UNTIL exactly at the last recurrence and then run the test with a TZ with positive offset and with one with a negative offset.

Reply:
The 2 arguments to `MatchTimeZone(IDateTime dt1, IDateTime dt2)` always have the same timezone, so there is nothing to adjust?
```C#
if (r.Until != DateTime.MinValue)
{
    r.Until = MatchTimeZone(referenceDate, new CalDateTime(r.Until, referenceDate.TzId)).Value;
}
```

Reply:
You are fully right, its a noop. But I strongly believe that something's wrong there, because the conversion needs to be done somewhere. Should be investigated separately. Maybe the translation between UTC and tzid happened in the constructor of CalDateTime and was lost in one of the previous PRs?

Reply:
Created #663 to track